### ISSUE TITLE
Write Image as PDF with Box includes text "Loading [MathJax]/extensions/MathMenu.js"

### ISSUE THREAD

**Issue by vincentliuheyang** [plotly/plotly.py#3469]
Hi,

<img width="892" alt="截屏2021-11-15 下午4 34 46" src="https://user-images.githubusercontent.com/49622643/141748703-aec34b85-5cd6-4423-ae19-6a422eaace47.png">

I am trying to write plot into PDF format, but I cannot get rid of this box saying "Loading [MathJax]/extensions/MathMenu.js", will you be able to tell me how should I fix it?

Here is the code I am using to generate the plot.
```
fig = go.Figure()
    fig.add_trace(go.Scatter(x=[0, 1, 2, 3, 4, 5], y=[0.5, 0.6, 0.7, 0.8, 0.9, 0.9], name='A',
                             line=dict(width=4), marker=dict(size=15, symbol='star')))
    fig.add_trace(go.Scatter(x=[0, 1, 2, 3, 4, 5], y=[0.4, 0.4, 0.4, 0.4, 0.4, 0.4], name='B',
                             line=dict(width=4), marker=dict(size=15, symbol='circle')))
    fig.add_trace(
        go.Scatter(x=[0, 1, 2, 3, 4, 5], y=[0.51, 0.52, 0.53, 0.54, 0.55, 0.56], name='C',
                   line=dict(width=4), marker=dict(size=15, symbol='x')))

    fig.update_layout(
        yaxis_range=[0.2, 1],
        legend=dict(
            x=0,
            y=0,
            traceorder="reversed",
            title_font_family="Times New Roman",
            font=dict(
                family="Courier",
                size=16,
                color="black"
            ),
            bgcolor="LightSteelBlue",
            bordercolor="Black",
            borderwidth=2
        )
    )
    fig.write_image("1.pdf")
    fig.show()
```

**Comment by empet** [comment]
What Plotly version are you running? I checked your code with   v 5.3.1 (the last one), and no such box is displayed.

**Comment by kuppu** [comment]
I am getting the same box in PDF outputs with Plotly Version 5.3.1
![image](https://user-images.githubusercontent.com/1645225/141809725-e50012b6-e74f-4cd9-98b4-5a17f0e13219.png)

**Comment by empet** [comment]
I have no idea why that box is displayed. It could be  your browser?!!!

**Comment by vincentliuheyang** [comment]
It only outputs such box if I am trying to write a PDF, the browser does not have such box. One peculiar issue is that if I am going to write multiple PDF plot in one file, this box will only show in the first PDF image.

**Comment by nicolaskruchten** [comment]
This is pretty unfortunate. The cause is that the PDF is being generated by a browser essentially doing a print operation. If this happens while MathJax is still loading, then this box will appear in the output. @jonmmease any thoughts about how to wait a bit longer?

**Comment by vincentliuheyang** [comment]
> This is pretty unfortunate. The cause is that the PDF is being generated by a browser essentially doing a print operation. If this happens while MathJax is still loading, then this box will appear in the output. @jonmmease any thoughts about how to wait a bit longer?

Just for now, my solution is writing a random plot before my official plot. Of course, I would like to see this problem resolved if possible.

**Comment by stansotn** [comment]
I am having the same issue with plotly 5.4.0 and kaleido 0.2.1.post1

**Comment by willfindlay** [comment]
I'm also running into this with:
```
plotly==5.4.0
plotly-express==0.4.1
kaleido==0.2.1
```
Thanks for the workaround @vincentliuheyang, it works well for me.

**Comment by jlehrer1** [comment]
Same issue with exact same versions as @willfindlay. Is there any workaround?

**Comment by tahashmi** [comment]
@jlehrer1 Try something like this :-) :

```
import plotly.express as px

#garbage graph
fig = px.scatter(x=[0, 1, 2, 3, 4], y=[0, 1, 4, 9, 16])
fig.show()
fig.write_image("random.pdf")

#actual graph to be replaced
fig1 = px.scatter(x=[0, 1, 2, 3, 4], y=[0, 1, 4, 9, 16])
fig1.show()
fig1.write_image("random.pdf")
```

**Comment by jorgeboucas** [comment]
this worked for me:

```
figure="some_figure.pdf"
fig=px.scatter(x=[0, 1, 2, 3, 4], y=[0, 1, 4, 9, 16])
fig.write_image(figure, format="pdf")
time.sleep(2)

fig=make_figure(df,pa)
fig.write_image(figure, format="pdf")
```

**Comment by jlehrer1** [comment]
Thank you! I hadn't realized it was only on the first write when running a script. Interesting.

**Comment by eric-vader** [comment]
I am experiencing the same problem as well, do we know what causes the problem?

**Comment by nicolaskruchten** [comment]
Yes, we know the basic cause, see https://github.com/plotly/Kaleido/issues/122 for details.

Note that if the figure you're rendering doesn't actually use MathJax then you can prevent MathJax from being loaded at all by Kaleido with:

```python
import plotly.io as pio   
pio.kaleido.scope.mathjax = None
```

Otherwise, if saving a "throwaway" PDF first helps, I would recommend that: basically save your figure twice.

**Comment by nicolaskruchten** [comment]
Is anyone seeing this with Kaleido 0.1 ?

**Comment by eric-vader** [comment]
I have tried to manually downgrade to kaleido 0.1 and plot my graph and the error still exist.

**Comment by Willtl** [comment]
A easy way around is to save as `.svg` and then export it as `.pdf` using Inkscape.

**Comment by pgvaz** [comment]
I have found a solution for that problem, at leas in my case it worked. Just disable the warnings on kaleido engine like this:

```
import plotly.io as pio

pio.full_figure_for_development(fig, warn=False)
fig.write_image("fig1.pdf",engine="kaleido")
```

**Comment by theo-brown** [comment]
Saving a dummy figure before the desired figure doesn't work for me.
What browser does Plotly use by default for PDF saving? Could it be a browser-dependent issue?

**Comment by zhimin-z** [comment]
> Yes, we know the basic cause, see [plotly/Kaleido#122](https://github.com/plotly/Kaleido/issues/122) for details.
> 
> Note that if the figure you're rendering doesn't actually use MathJax then you can prevent MathJax from being loaded at all by Kaleido with:
> 
> ```python
> import plotly.io as pio   
> pio.kaleido.scope.mathjax = None
> ```
> 
> Otherwise, if saving a "throwaway" PDF first helps, I would recommend that: basically save your figure twice.

Hi, I tried but it did remove the watermark, but fails to render the math any more. I need the math formula to be shown, what can I do now? @nicolaskruchten

**Comment by InvictusRMC** [comment]
I just experienced the same issue, my workaround is calling `show` after `write_image`.

**Comment by Fratorhe** [comment]
I am having the same issue, I have tried in Windows 10, WSL and Ubuntu...
I am using the most recent versions of the libraries. 
An option could be to export to svg and then convert with inkscape (I tried with cairosvg and it does not render the mathjax). 
Any updates or alternatives?

**Comment by PatrickRWright** [comment]
I am getting this when running in `R` via `plotly::save_image("afile.pdf")`.

**Comment by emilykl** [comment]
plotly.py v7.0.0 now bundles MathJax v3 rather than v2 (see https://github.com/plotly/plotly.py/pull/5684), and the message bar is [no longer included in v3](https://docs.mathjax.org/en/v4.0/upgrading/v2.html#no-longer-applies-to-version-3), so this is resolved.

Note that Kaleido [v1.4.0](https://github.com/plotly/Kaleido/releases/tag/v1.4.0) or greater must be used for generating images, since that is the first version which uses MathJax v3 by default.
